infinite-tracing/src/main/java/com/newrelic/ChannelSupplier.java

```diff
@@ -12,47 +12,42 @@
  * Not thread-safe.
  */
 public class ChannelSupplier implements Supplier<ManagedChannel> {
-    private final ConnectionStatus connectionStatus;
+    private final ConnectionSynchronizer connectionSynchronizer;
     private final Logger logger;
     private final ChannelFactory channelFactory;
     private volatile ManagedChannel channel;
 
-    public ChannelSupplier(ConnectionStatus connectionStatus, Logger logger, ChannelFactory channelFactory) {
-        this.connectionStatus = connectionStatus;
+    public ChannelSupplier(ChannelFactory channelFactory, Logger logger, ConnectionSynchronizer connectionSynchronizer) {
+        this.connectionSynchronizer = connectionSynchronizer;
         this.logger = logger;
         this.channelFactory = channelFactory;
         channel = null;
     }
 
     @Override
     public ManagedChannel get() {
-        ConnectionStatus.BlockResult blockResult = getBlockResult();
+        boolean requiresReconnect = awaitRequiresReconnect();
 
-        if (blockResult == ConnectionStatus.BlockResult.GO_AWAY_FOREVER) {
-            throw new RuntimeException("No longer attempting to connect.");
-        }
```

Comment:
Generating this exception is now handled within `ConnectionSynchronizer`. This enables the response to be simplified from an 4 value enum to a boolean, which makes it easier to reason about what a caller should do.